fix: improved elastic search query filter

Junjiequan opened the pull request:
## Description

This PR contains following fixes and improves:

**Fixes:**

- Fixed ScientificMetadata condition filter. Before the change, filter does not work with floating number, due to incorrect mapping settings for elastic search.

**Improves:**

- Included fallback for elastic search. Before the change, the datasets table is empty if ES DB is not synched with mongoDB. Now, the dataset full query endpoint checks if ES query returns any result, if ES query does not return result then it uses regular mongo query.


## Motivation

Background on use case, changes needed

## Fixes:

* Items added

## Changes:

* changes made

## Tests included/Docs Updated?

- [ ] Included for each change/fix?
- [ ] Passing? (Merge will not be approved unless this is checked) 
- [ ] Docs updated?
- [ ] New packages used/requires npm install? 
- [ ] Toggle added for new features?

Junjiequan committed:
fix: updated mathJS package version && improved number check

Junjiequan committed:
fix: improved elastic search query filter

Junjiequan committed:
fix: fixed inconsistency between data list and count number

Junjiequan committed:
fix: minor naming change

Junjiequan committed:
fix: improved mapping for digit with floating points

Junjiequan committed:
Merge branch 'master' into SWAP-3608-scicat-datasets-table-total-number-and-actual-i

Junjiequan committed:
fix: revert back to mongo query if es db doesnt return any value

bpedersen2 reviewed (changes requested):
Except for some  minor comments, looks fine.

Junjiequan commented:
@bpedersen2  Could you copy paste the comments here? I'm not able to see the comments somehow

dependabot[bot] committed:
build(deps): bump actions/setup-node from 3 to 4

Bumps [actions/setup-node](https://github.com/actions/setup-node) from 3 to 4.
- [Release notes](https://github.com/actions/setup-node/releases)
- [Commits](https://github.com/actions/setup-node/compare/v3...v4)

---
updated-dependencies:
- dependency-name: actions/setup-node
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

github-actions[bot] committed:
Merge pull request #855 from SciCatProject/dependabot/github_actions/actions/setup-node-4

build(deps): bump actions/setup-node from 3 to 4

Junjiequan committed:
fix: fixed pagination conflict between ES and mongo query

Junjiequan committed:
fix: remove empty space in string value of condition filter with trim()

bpedersen2 commented on src/elastic-search/configuration/indexSetting.ts:
```diff
@@ -39,12 +39,12 @@ export const dynamic_template:
       path_match: "scientificMetadata.*.*",
```
long_as_double??

Junjiequan committed:
fix: minor change for object name and removed redundant type

Junjiequan committed:
Merge branch 'master' into SWAP-3608-scicat-datasets-table-total-number-and-actual-i

bpedersen2 commented:
needs local rebase and force-pushing to `SWAP-3608-scicat-datasets-table-total-number-and-actual-i` branch
